ui/core/sim_concurrent.ts

```diff
@@ -91,12 +92,8 @@ function runSims(requests: RaidSimRequest[], totalIterations: number, wp: Worker
 				if (pm.finalRaidResult.errorResult) {
 					console.error(`Worker ${idx} had an error!`);
 					errRes = pm.finalRaidResult;
-					// This sucks, but it's better than having long running workers forever.
-					if (requests[0].simOptions!.iterations > 1000) {
-						console.log("Terminating all workers to get going again.");
-						const num = wp.getNumWorkers();
-						wp.setNumWorkers(0);
-						wp.setNumWorkers(num);
+					for (const req of requests) {
+						wp.abortById(req.id);
```

Comment:
I would do this to make sure and await all requests in parallel before resolving, or is that not needed?:
```
await Promise.all(requests.map(request => wp.abortById(req.id)))
```

Reply:
This is just to stop still running workers if only one crashed for some reason. Only the first error is passed on, what the other workers do doesn't matter anymore at that point.